### PR TITLE
[MOD-2268] Support for the Visible property + removing dummy code.

### PR DESCRIPTION
**Pull request by rodperottoni**
Android - Support for the "Visible" property and removing some dummy code.

JIRA: https://jira.appcelerator.org/browse/MOD-2268

### CODE REVIEW

**Comment by hansemannn** [comment]
Hey there, do you have a JIRA ticket for this improvement? Then we could review it

**Comment by rodperottoni** [comment]
Hey Hans. Here it goes: https://jira.appcelerator.org/browse/AC-4044

**Comment by cheekiatng** [comment]
Thanks @rodperottoni ! Btw there are merge conflicts do you think you can fix it?

**Comment by hansemannn** [comment]
@rodperottoni ping

File: android/src/ti/map/AnnotationProxy.java [hunk 1]
```diff
@@ -70,10 +71,12 @@
 	private static final int MSG_SET_LON = MSG_FIRST_ID + 300;
 	private static final int MSG_SET_LAT = MSG_FIRST_ID + 301;
 	private static final int MSG_SET_DRAGGABLE = MSG_FIRST_ID + 302;
-	private static final int MSG_UPDATE_INFO_WINDOW = MSG_FIRST_ID + 303;
+	private static final int MSG_SET_VISIBLE = MSG_FIRST_ID + 303;
```
Comment (hansemannn) [inline android/src/ti/map/AnnotationProxy.java]: Although it shouldn't really matter in this context, I would prefer to keep `MSG_UPDATE_INFO_WINDOW` the 303 and make `MSG_SET_VISIBLE` the 304.

File: android/src/ti/map/AnnotationProxy.java [hunk 2]
```diff
@@ -340,33 +356,22 @@ public void onPropertyChanged(String name, Object value)
...
 			getOrCreateMapInfoWindow().setLeftOrRightPane(value, TiMapInfoWindow.LEFT_PANE);
 			updateInfoWindow();
 		} else if (name.equals(TiC.PROPERTY_RIGHT_BUTTON)) {
 			getOrCreateMapInfoWindow().setLeftOrRightPane(value, TiMapInfoWindow.RIGHT_PANE);
-			if (value == null) {
-				Object rightView = getProperty(TiC.PROPERTY_RIGHT_VIEW);
-				if (rightView != null) {
-					getOrCreateMapInfoWindow().setLeftOrRightPane(rightView, TiMapInfoWindow.LEFT_PANE);
-				}
-			}
```
Comment (hansemannn) [inline android/src/ti/map/AnnotationProxy.java]: Why did you remove `rightView`?

**Comment by hansemannn** [comment]
Left some comments, please address and rebase (first). Also, we need docs for this new property and parity with iOS, I could help with that.

File: android/src/ti/map/AnnotationProxy.java [hunk 3]
```diff
@@ -340,33 +356,22 @@ public void onPropertyChanged(String name, Object value)
 			updateInfoWindow();
 		} else if (name.equals(TiC.PROPERTY_LEFT_BUTTON)) {
 			getOrCreateMapInfoWindow().setLeftOrRightPane(value, TiMapInfoWindow.LEFT_PANE);
-			if (value == null) {
-				Object leftView = getProperty(TiC.PROPERTY_LEFT_VIEW);
-				if (leftView != null) {
-					getOrCreateMapInfoWindow().setLeftOrRightPane(leftView, TiMapInfoWindow.LEFT_PANE);
-				}
-			}
```
Comment (rodperottoni) [inline android/src/ti/map/AnnotationProxy.java]: Hey Hans. I've removed it because it has it's own conditional on line 360. It was duplicated.

File: android/src/ti/map/AnnotationProxy.java (hunk 2 above)
Comment (rodperottoni) [inline android/src/ti/map/AnnotationProxy.java]: Same as the other one, this property has its own conditional at line 366.

**Comment by rodperottoni** [comment]
Please cancel this Pull Request. It would take me too much time to rebase my fork to the state where the official one already is. I'll create another pull request.

**Comment by rodperottoni** [comment]
(actually i screwed up the rebase - i'm a ignorant on rebasing). Better be safe than sorry.
